### DISCUSSION TITLE
Page does not work using Custom CacheHash

### DISCUSSION THREAD

**Discussion by dLopes-SE** [SixLabors/ImageSharp.Web#174] [Q&A]
Hello!

I'm using a custom CacheHash class (ICacheHash's successor) to create custom keys for the caching blobs.

```csharp
public interface ICustomCacheHash : ICacheHash
{
    new string Create(string value, uint length);
}
````

```csharp
public class CustomCacheHash : ICustomCacheHash
{
    public CustomCacheHash(IOptions<ImageSharpMiddlewareOptions> options)
    {

    }

    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    public string Create(string value, uint length)
    {
        Uri url = new Uri(value);
        string image = url.AbsolutePath.Split('/').Last();

        if (value.Contains("quality"))
        {                
            string format = HttpUtility.ParseQueryString(url.Query).Get("size");
            return image.Split('.')[0] + "_" + format + "." + image.Split('.')[1];
        }

        else
        { 
            return image.Split('.')[0] + "_w" + HttpUtility.ParseQueryString(url.Query).Get("width") + "." + image.Split('.')[1];
        }
    }
}
```

This actually works great locally. However, once I publish to an Azure domain, the page doesn't work.

Note: The page works when I publish using the original Classes.

**Comment by JimBobSquarePants** [comment]
Have you debugged it?

**Reply by dLopes-SE** [reply]
I did it locally, and it works perfectly. On the other hand, I didn't try to debug in prod. Shouldn't the Model folder (where I got the custom classes) be included in the app dll, which goes to prod during the publish?

**Reply by JimBobSquarePants** [reply]
As log as you've not instructed the project to not include the file in the compilation. You can check this by inspecting the csproj. If you are still unsure you can use various tool to inspect the compiled binary itself.

If it works locally then this is a build issue and any assistance is out of scope here.

**Reply by dLopes-SE** [reply]
I understand. Thanks for your help.
